components/toast_wrapper/toast_wrapper.jsx

```diff
@@ -120,31 +130,45 @@ class ToastWrapper extends React.PureComponent {
             showNewMessagesToast = true;
         }
 
+        if (props.unreadScrollPosition === Preferences.UNREAD_SCROLL_POSITION_START_FROM_NEWEST && !props.channelMarkedAsUnread) {
+            showUnreadToast = false;
+        }
+
         if (!unreadCount) {
-            showNewMessagesToast = false;
             showUnreadToast = false;
+            showNewMessagesToast = false;
+        }
+
+        if (props.shouldStartFromBottomWhenUnread &&
+            unreadCount > 0 &&
+            (props.lastViewedBottom >= props.latestPostTimeStamp) &&
+            prevState.atBottom === null && props.atBottom
+        ) {
+            showUnreadWithBottomStartToast = true;
```

Comment:
Just to make sure I understand this, we want to show the "scroll up to unreads" toast if we loaded the channel at the bottom and the channel is unread, is that correct? I'm not entirely sure what the 3rd and fourth lines of the if condition are doing

Reply:
I was intended to show the "scroll up to unreads" toast if scroll is at the bottom, but I found checking these two lines are unnecessary. I removed these lines.

+) I added a line that checks `props.shouldHideNewMessageIndicator` to hide the toast when scroll reached  the unread posts.